src/core/settings/qgssettings.h

```diff
@@ -233,6 +264,13 @@ class CORE_EXPORT QgsSettings : public QObject
      */
     void setArrayIndex( int i );
 
+    /**
+     * Returns the location of the setting if it exists at the given \a key
+     * \note it will return Qgis::SettingsLocation::Any if the key doesn't exist
+     * \since QGIS 3.30
+     */
+    Qgis::SettingsLocation location( const QString &key ) const;
```

Comment:
Here we have `QString`, in qgssettingsentry.cpp a `QStringList`, is this on purpose?

Reply:
QgsSettingsEntry knows about its key. The location method has an optional parameters for the dynamic parts (as a string list)
Potentially, I could add another method with a QString as argument if there only 1 dynamyc part, but I am trying to avoid adding too many methods.

in QgsSettings, we give the key of the settings to test.